torch/csrc/jit/codegen/cuda/tensor_view.cpp

```
@@ -391,6 +391,43 @@ TensorView* TensorView::rFactor(const std::vector<int>& axes) {
   return producer;
 }
 
+std::vector<TensorView*> TensorView::duplicate() {
```

If this is an input or output we should throw an error.

---

This interface seems strange for "recomputing" a tensor as it doesn't recompute the entire graph producing a TensorView, only a single expression.

---

Should duplicated tensors be replayed as the tensor they duplicated? What about logic like computeAt of the duplicate?

---

Do you want a single TensorView op to manipulate the entire graph?

---

I call `TransformReplay::fullSelfReplay` to transform the duplicated tensor.